ratatui-core/src/symbols/line.rs

```diff
@@ -58,6 +58,270 @@ pub const CROSS: &str = "┼";
 pub const DOUBLE_CROSS: &str = "╬";
 pub const THICK_CROSS: &str = "╋";
 
+#[derive(PartialEq, Clone)]
+pub enum LineStyle {
+    Nothing,
+    Plain,
+    Rounded,
+    Double,
+    Thick,
+    DoubleDash,
+    TripleDash,
+    TripleDashThick,
+    QuadrupleDash,
+    QuadrupleDashThick,
+}
+
+impl LineStyle {
+    fn replace(self, from: &Self, to: &Self) -> Self {
+        if self == *from {
+            to.clone()
+        } else {
+            self
+        }
+    }
+}
+
+pub struct BorderSymbol {
+    pub right: LineStyle,
+    pub up: LineStyle,
+    pub left: LineStyle,
+    pub down: LineStyle,
+}
+
+impl BorderSymbol {
```

Comment:
Move `new` from the other impl block, add comments.

```suggestion
impl BorderSymbol {
    /// Creates a new `BorderSymbol`, based on individual line styles.
    #[must_use]
    pub const fn new(right: LineStyle, up: LineStyle, left: LineStyle, down: LineStyle) -> Self {
        Self {
            right,
            up,
            left,
            down,
        }
    }
```